core/src/main/scala/chisel3/properties/Property.scala

```diff
@@ -298,29 +314,58 @@ sealed trait Property[T] extends Data { self =>
 
 }
 
+trait ClassTypeProvider[A] {
+  val classType: fir.PropertyType
+}
+
+object ClassTypeProvider {
+  def apply[A](className: String) = new ClassTypeProvider[A] {
+    val classType = fir.ClassPropertyType(className)
+  }
+  def apply[A](_classType: fir.PropertyType) = new ClassTypeProvider[A] {
+    val classType = _classType
+  }
+}
+
+trait Typeable[T, Lit] {
```

Comment:
```suggestion
sealed trait Typeable[T, Lit] {
```
This should be sealed right?

Reply:
deleted this, ended up not needing it